### PR TITLE
Fix/nested domain test

### PR DESCRIPTION
**Pull request by Kunal-kankriya**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js [hunk 1]
```diff
@@ -1,53 +1,93 @@
...
 
-        // Move a domain from under a parent domain to the root level
-        cy.get('[data-testid="domain-list-item"]').contains("Marketing").prev().click();
-        cy.clickOptionWithText(domainName);
-        cy.openThreeDotDropdown();
-        cy.clickOptionWithTestId("entity-menu-move-button");
-        cy.clickOptionWithTestId("move-domain-modal-move-button").wait(5000);
+    it("Move domain parent level to root level", () => {
+        moveDomaintoParent();
+        cy.waitTextVisible("Moved Domain!")
```
Comment (anshbansal) [inline smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js:81]: This looks like something that should be in `moveDomaintoParent` instead

File: smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js [hunk 2]
```diff
@@ -1,53 +1,93 @@
...
+    it("Create a new domain", () => {
+        deleteExisitingDomain()      
+        createDomain();
+        cy.waitTextVisible("Created domain!");
+        cy.waitTextVisible("Domains");
+    });
+        
+    it("Move domain root level to parent level", () => {
+        moveDomaintoRootLevel();
+        cy.waitTextVisible("Moved Domain!")
```
Comment (anshbansal) [inline smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js:74]: This looks like something that should be in `moveDomaintoParent` instead

File: smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js [hunk 3]
```diff
@@ -1,53 +1,93 @@
...
+            deleteFromDomainDropdown();
+        }
+    });   
+    cy.waitTextVisible('My custom domain'); 
+ }   
+
+describe("nested domains test", () => {
+  
+    beforeEach (() => {
+    cy.loginWithCredentials();
```
Comment (anshbansal) [inline smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js:61]: please fix indentation

File: smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js [hunk 4]
```diff
@@ -1,53 +1,93 @@
...
+          .then((length) => {
+        for (let i = 0; i < length - 1; i++) {
+            cy.get('a[href*="urn:li"] span[class^="ant-typography"]')
+                .should('be.visible')
+                .first()
+                .click({ force: true });
+            deleteFromDomainDropdown();
+        }
+    });   
+    cy.waitTextVisible('My custom domain');
```
Comment (anshbansal) [inline smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js:55]: Why is this supposed to be here at the end of deleting all domains?

File: smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js [hunk 5]
```diff
@@ -1,53 +1,93 @@
...
+  
+    beforeEach (() => {
+    cy.loginWithCredentials();
+    cy.goToDomainList();
+  });
+  
+    it("Create a new domain", () => {
+        deleteExisitingDomain()      
+        createDomain();
+        cy.waitTextVisible("Created domain!");
```
Comment (anshbansal) [inline smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js:68]: Should this be in `createDomain`?

File: smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js (hunk 4 above)
Comment (Kunal-kankriya) [reply smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js:55]: It's a validation that waits for the text to become visible. 
The text pertains to the Marketing domain, 
which is a single domain already present in domain list
and remains unchanged during the test execution.

File: smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js (hunk 5 above)
Comment (Kunal-kankriya) [reply smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js:68]: yes

File: smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js (hunk 3 above)
Comment (Kunal-kankriya) [reply smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js:61]: sure

**Review by anshbansal — APPROVED** [review]

**Pull request merged**